### PR TITLE
actions: add automatic labeler action

### PR DESCRIPTION
**Pull request by nashif**
Add action to labler pull requests on a regular basis. This will replace
the current labler implemented in ci-tools and will operate on a
schedule rather than on pull request trigger.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+"area: NIOS2":
+  - "arch/nios2/**/*"
+"area: Xtensa":
+  - "arch/xtensa/**/*"
+"area: RISCv32/64":
+  - "arch/risv/**/*"
+"area: ARC":
+  - "arch/arc/**/*"
+"area: Networking":
+  - "subsys/net/**/*"
```
Comment (jukkar) [inline .github/labeler.yml:47]: we could add here
`- "include/net/**/*"`
`- "include/drivers/ieee802154/**/*"`
`- "drivers/ethernet/**/*"`
`- "drivers/ieee802154/**/*"`
`- "drivers/wifi/**/*"`
`- "drivers/ptp_clock/**/*"`
`- "drivers/net/**/*"`

File: .github/labeler.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  - "samples/net/**/*"
+  - "tests/net/**/*"
+	- "include/net/**/*"
+	- "include/drivers/ieee802154/**/*"
+	- "drivers/ethernet/**/*"
+	- "drivers/ieee802154/**/*"
+	- "drivers/wifi/**/*"
+	- "drivers/ptp_clock/**/*"
+	- "drivers/net/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:56]: tabs

File: .github/labeler.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  - "drivers/adc/**/*"
+"area: Counter":
+  - "drivers/counter/**/*"
+"area: Timer":
+  - "drivers/timer/**/*"
+"area: I2S":
+  - "drivers/i2s/**/*"
+"area: C Library":
+  - "lib/libc/**/*"
+"area: Deviec Tree":
```
Comment (stephanosio) [inline .github/labeler.yml:19]:
```suggestion
"area: Device Tree":
```

File: .github/labeler.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+"area: POSIX":
+  - "lib/posix/**/*"
+"area: native port":
+  - "arch/posix/**/*"
+  - "soc/posix/**/*"
+  - "**/*native_posix*"
+"area: X86":
+  - "arch/x86/**/*"
+"area: ARM":
+  - "arch/arm/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:37]: Should add the corresponding arch include dirs for all arch labels
```suggestion
"area: ARM":
  - "arch/arm/**/*"
  - "include/arch/arm/**/*"
```

File: .github/labeler.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+"area: Boards":
+  - "boards/**/*"
+"area: POSIX":
+  - "lib/posix/**/*"
+"area: native port":
+  - "arch/posix/**/*"
+  - "soc/posix/**/*"
+  - "**/*native_posix*"
+"area: X86":
+  - "arch/x86/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:35]:
```suggestion
  - "arch/x86/**/*"
  - "include/arch/x86/**/*"
```

File: .github/labeler.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+"area: ARM":
+  - "arch/arm/**/*"
+  - "include/arch/arm/**/*"
+"area: NIOS2":
+  - "arch/nios2/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:40]:
```suggestion
  - "arch/nios2/**/*"
  - "include/arch/nios2/**/*"
```

File: .github/labeler.yml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+"area: Xtensa":
+  - "arch/xtensa/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:42]:
```suggestion
  - "arch/xtensa/**/*"
  - "include/arch/xtensa/**/*"
```

File: .github/labeler.yml [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+"area: RISCv32/64":
+  - "arch/risv/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:44]:
```suggestion
  - "arch/risv/**/*"
  - "include/arch/riscv/**/*"
```

File: .github/labeler.yml [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+"area: ARC":
+  - "arch/arc/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:46]:
```suggestion
  - "arch/arc/**/*"
  - "include/arch/arc/**/*"
```

File: .github/labeler.yml [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+"area: I2C":
+  - "drivers/i2c/**/*"
+"area: SPI":
+  - "drivers/spi/**/*"
+"area: Boards":
+  - "boards/**/*"
+"area: POSIX":
+  - "lib/posix/**/*"
+"area: native port":
+  - "arch/posix/**/*"
```
Comment (stephanosio) [inline .github/labeler.yml:31]:
```suggestion
  - "arch/posix/**/*"
  - "include/arch/posix/**/*"
```

File: .github/labeler.yml [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+  - "subsys/testsuite/**/*"
+"area: Settings":
+  - "subsys/settings/**/*"
+"area: File System":
+  - "subsys/fs/**/*"
+"area: Storage":
+  - "subsys/storage/**/*"
+"area: Bluetooth":
+  - "subsys/bluetooth/**/*"
+  - "**/*bluetooth*"
```
Comment (jhedberg) [inline .github/labeler.yml:74]: I'd still add include/bluetooth include/drivers/bluetooth drivers/bluetooth samples/bluetooth and tests/bluetooth to this list.

File: .github/labeler.yml [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+  - "drivers/counter/**/*"
+"area: Timer":
+  - "drivers/timer/**/*"
+"area: I2S":
+  - "drivers/i2s/**/*"
+"area: C Library":
+  - "lib/libc/**/*"
+"area: Device Tree":
+  - "dts/**/*"
+  - "**/*.dts"
```
Comment (erwango) [inline .github/labeler.yml:21]:
```suggestion
  - "**/*.dts"
  - "include/devicetree/*"
  - "include/dt-bindings/**/*"
```

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by nashif** [comment]
Note, I am just copying the existing mapping, I know there are many missing, but this should not block this from going in and have additional items added later. If we are to include every path in this PR, this will probably never be merged in time :)

**Review by erwango — APPROVED** [review]

File: .github/labeler.yml (hunk 11 above)
Comment (nashif) [reply .github/labeler.yml:82]: those will already be covered with the second rule...

**Review by jhedberg — APPROVED** [review]

**Pull request merged**